python/pyspark/sql/tests.py

```diff
@@ -429,6 +429,17 @@ def test_udf_with_input_file_name(self):
         row = self.spark.read.json(filePath).select(sourceFile(input_file_name())).first()
         self.assertTrue(row[0].find("people1.json") != -1)
 
+    def test_udf_with_decorator(self):
+        from pyspark.sql.decorators import udf
+        from pyspark.sql.types import IntegerType
+
+        @udf(IntegerType())
```

Comment:
Could you add a test case for the default return type? I think it currently requires empty-parens:

```
@udf()
def truncate(s):
    return s[:10]
```